Kitodo/src/main/java/org/kitodo/production/forms/ProzesskopieForm.java

```diff
@@ -498,7 +511,31 @@ protected void removeCollectionsForChildren(LegacyMetsModsDigitalDocumentHelper
      * Create the process and save the metadata.
      */
     public String createNewProcess() {
+        if (Objects.nonNull(titleRecordLinkTab.getTitleRecordProcess())) {
+            if (Objects.isNull(titleRecordLinkTab.getInsertionPositionSelectionSelectedItem())) {
+                Helper.setErrorMessage("prozesskopieForm.createNewProcess.noInsertionPositionSelected");
+                return stayOnCurrentPage;
+            } else {
+                User titleRecordOpenUser = DataEditorForm
+                        .getUserOpened(titleRecordLinkTab.getTitleRecordProcess().getId());
+                if (Objects.nonNull(titleRecordOpenUser)) {
+                    Helper.setErrorMessage("prozesskopieForm.createNewProcess.titleRecordOpen",
+                        titleRecordOpenUser.getFullName());
+                    return stayOnCurrentPage;
+                }
+            }
+        }
         if (createProcess()) {
+            if (Objects.nonNull(titleRecordLinkTab.getTitleRecordProcess())) {
+                ServiceManager.getProcessService().refresh(prozessKopie);
+                try {
+                    MetadataEditor.addLink(titleRecordLinkTab.getTitleRecordProcess(),
+                        titleRecordLinkTab.getInsertionPositionSelectionSelectedItem(), prozessKopie.getId());
+                } catch (IOException | RuntimeException exception) {
+                    Helper.setErrorMessage("errorSaving", titleRecordLinkTab.getTitleRecordProcess().getTitle(), logger,
+                        exception);
```

Comment:
Good question. At first I implemented it that way, but it's also unintuitive if the object was created in the database, and then you stay on the page and then get an error saving that the object already exists. If you take it very carefully, the object would have to be deleted from the database at that point. Then because of the now assigned ID, the form would have to be reinitialized and the data copied over. Then the user should stay on the page. Since the case of an error occurring here is extremely unlikely, we should first discuss what is necessary here before investing too much work.